entity/contents/linear_gradient_contents.cc

```diff
@@ -46,6 +48,8 @@ std::optional<Rect> LinearGradientContents::GetCoverage(
 bool LinearGradientContents::Render(const ContentContext& renderer,
                                     const Entity& entity,
                                     RenderPass& pass) const {
+  FML_DCHECK(path_set_) << "Render was called without setting a path.";
```

Comment:
Here and elsewhere:

Can we just check that the path is not empty (`FML_DCHECK(path_.GetComponentCount() > 0)`)? And then guard upstream so that even if Flutter app asks us to draw an empty path, we short circuit before we get to rendering contents?

That eliminates a weird bool that we're likely to forget to set somewhere, and that doesn't quite cover validity for contents.

Reply:
Ehhh, yeah the extra state is ugly -- I reversed it. Thinking this through, I'm not so sure we really need any checks here. I think just having `EntityPass` print a message that says the render failed because of an entity failing would have made this issue easy enough to track down.

The original checks help to prevent a developer error, but having an empty path supplied here is not an error and shouldn't cause a failure here after the changes I made to the tessellator.

Reply:
I'm of a mind to say that we should make rendering an empty path an error at this point in the stack, but that could be hashed out in a separate PR.